### ISSUE TITLE
what's the corresponding method of these functions in codes

### ISSUE THREAD

**Issue by MaKiTann** [yihui-he/channel-pruning#46]
I found that there is a lot function names that are hard to tell which method it uses
for example 
the "c3" in train.py which indicates these three methods

1.  Speeding up Convolutional Neural Networks with Low Rank Expansions
2.  Accelerating Very Deep Convolutional Networks for Classification and Detection
3.  Channel Pruning for Accelerating Very Deep Neural Networks, the Lasso one

but which function implements these methods respectively?

Here is the list of function that I can't realize the corresponding method in papers
1. YYT (https://github.com/yihui-he/channel-pruning/blob/master/lib/decompose.py#L61)
2. VH_decompose (https://github.com/yihui-he/channel-pruning/blob/master/lib/decompose.py#L85)
3. ITQ_decompose (https://github.com/yihui-he/channel-pruning/blob/master/lib/decompose.py#L163)
4. dictionary (https://github.com/yihui-he/channel-pruning/blob/master/lib/decompose.py#L386)
5. fc_kernel (https://github.com/yihui-he/channel-pruning/blob/master/lib/decompose.py#L636)
6. nonlinear_fc (https://github.com/yihui-he/channel-pruning/blob/master/lib/decompose.py#L671)

Would you explain these functions?
Thanks!

**Comment by slothkong** [comment]
I've been reading the code for two weeks now, and my understanding is that VH_decompose() performs spatial decomposition of the layers (splits them into 2 new layers with different shapes), ITQ _decompose() carries out channel decomposition @yihui-he  explained a bit in  issue #15. dictionary_kernel() is a function wrapper of dictionary(), which in charge of performing the lasso regression and minimizing the error reconstruction when selecting feature maps to prune. 

Honestly I don't know why is there references to fully connected layers in the code, but from what I have read, fc_kernel and nonlinear_fc are not used during the execution of train.py. I have added more comment to the code in a fork I made. Feel free to check it out.
